### PR TITLE
bpo-41111: Build limited C API without Py_TRACE_REFS macro.

### PR DESCRIPTION
**Pull request by shihai1991**
https://bugs.python.org/issue41111

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1864,9 +1864,10 @@ def detect_modules(self):
 ##         # Uncomment these lines if you want to play with xxmodule.c
 ##         self.add(Extension('xx', ['xxmodule.c']))
 
-        # Limited C API
-        self.add(Extension('xxlimited', ['xxlimited.c']))
-        self.add(Extension('xxlimited_35', ['xxlimited_35.c']))
+        # Build limited C API without Py_TRACE_REFS macro
```
Comment (vstinner) [inline setup.py]: This comment is wrong. You should say that the limited C API is not compatible with the Py_TRACE_REFS macro.

Comment (shihai1991) [reply setup.py]: Got it. Updated :)

**Comment by vstinner** [comment]
Thanks, I merged your fix.

Without this fix, xxlimited failed to build on the AMD64 Arch Linux TraceRefs 3.x buildbot. Example:

```
gcc -pthread -fPIC -Wno-unused-result -Wsign-compare -g -Og -Wall -std=c99 -Wextra -Wno-unused-result -Wno-unused-parameter -Wno-missing-field-initializers -Wstrict-prototypes -Werror=implicit-function-declaration -fvisibility=hidden -I./Include/internal -I./Include -I. -I/usr/local/include -I/buildbot/buildarea/3.x.pablogsal-arch-x86_64/build/Include -I/buildbot/buildarea/3.x.pablogsal-arch-x86_64/build -c /buildbot/buildarea/3.x.pablogsal-arch-x86_64/build/Modules/xxlimited_35.c -o build/temp.linux-x86_64-3.10-pydebug/buildbot/buildarea/3.x.pablogsal-arch-x86_64/build/Modules/xxlimited_35.o
In file included from ./Include/Python.h:87,
                 from /buildbot/buildarea/3.x.pablogsal-arch-x86_64/build/Modules/xxlimited_35.c:10:
./Include/object.h:61:4: error: #error Py_LIMITED_API is incompatible with Py_TRACE_REFS
   61 | #  error Py_LIMITED_API is incompatible with Py_TRACE_REFS
      |    ^~~~~
```

https://buildbot.python.org/all/#/builders/484/builds/525

But it's not treated as an error: the compilation is displayed as a success. Moreover, test_xxlimited is simply skipped if xxlimited is missing.

This change should fix it.

**Comment by shihai1991** [comment]
Thanks Victor for your review & merge :)

**Pull request merged**